### ISSUE TITLE
When a jQuery.deferred is passed to jQuery.deferred.resolve the value at chained .then() is the jQuery.Deferred object

### ISSUE THREAD

**Issue by guest271314** [jquery/jquery#3802]
Given 
```

  let promise = () => new Promise(resolve => resolve("native Promise"));

  new Promise((resolve, reject) => {
      setTimeout(() => reject("10 seconds exceeded"), 10000);
      resolve(promise())
    })
    .then(data => console.log(data))
    .catch(err => console.error(err));
```

"native Promise" is logged at `.then()` chained to `Promise` constructor.

Using same pattern with `jQuery.Deferred()` at version 3.2.1 the jQuery deferred object is logged at `.then()` chained to `jQuery.Deferred()`

  
```
  let promise = () => new $.Deferred(dfd => dfd.resolve("jQuery.Deferred promise"));

  new $.Deferred(dfd => {
      setTimeout(() => dfd.reject("10 seconds exceeded"), 10000);
      dfd.resolve(promise());
    })
    .then(data => console.log(data))
    .fail(err => console.error(err));
```
jsfiddle https://jsfiddle.net/r1gsLk4k/

- The expected result is `"jQuery.Deferred promise"` at chained `.then()`.

- The actual result is the `jQuery.Deferred` object at `.then()`.

- The result is the same at both Chromium 60 and Firefox 57

**Comment by gibson042** [comment]
For backwards compatibility with older versions of jQuery, [`Deferred#resolve`](http://api.jquery.com/deferred.resolve/) fulfills with values exactly as provided, and they will be supplied to registered callbacks unmodified. However, [`Deferred#then`](http://api.jquery.com/deferred.then/) is fully Promises/A+ compatible, so the Deferreds returned from _it_ will always unwrap their fulfillment values before providing them to callbacks.

In short, `$.Deferred().then(callback)` may receive a thenable, but `$.Deferred().then().then(callback)` never will: https://jsfiddle.net/r1gsLk4k/1/

**Comment by guest271314** [comment]
@gibson042 This means that `[Deferred#resolve](http://api.jquery.com/deferred.resolve/)` is not Promises/A+ compatible, correct?
